includes/Services/SiteGenService.php

```diff
@@ -253,6 +254,260 @@ public static function generate_child_theme( $data ) {
 
 	}
 
+	/**
+	 * Gets the preview homepages
+	 *
+	 * @return array
+	 */
+
+	public static function generate_homepages($site_description, $content_style, $target_audience, $regenerate = false) {
+        // Fetch homepages using the SiteGen::get_home_pages method
+        $home_pages = SiteGen::get_home_pages(
+            $site_description,
+            $content_style,
+            $target_audience,
+            $regenerate
+        );
+
+        // Process the homepages
+        $processed_home_pages = self::process_homepages_response($home_pages);
+
+        // Update the option with processed homepages
+        update_option('nfd-sitegen-homepages', $processed_home_pages);
+
+        return $processed_home_pages;
+    }
+
+	/**
+	 * Updates the list of favourited slugs
+	 *
+	 * @return array
+	 */
+	public static function toggle_favorite_homepage($slug) {
+        $favorites = get_option('nfd-sitegen-favorites', []);
```

Comment:
If we have an `isFavourited` option and we need to iterate through `nfd-sitegen-homepages` then why do we need `nfd-sitegen-favorites`?